design/hardware-classification-controller/support-for-new-parameters-hwcc.md

```diff
@@ -0,0 +1,268 @@
+# add support for extra hardware parameters in Disk, RAM, CPU and NIC
+
+## Status
+
+Implementable
+
+## Summary
+
+Implement support for more parameters in HWCC which will help to
+classify a baremetal host on more possible configurations.
+
+## Motivation
+
+We have hardware classification controller which validates and
+classifies the hosts based on user hardware requirements.
+We would like to add more parameters which helps user to
+identify the ideal host. This feature helps user to avoid runtime failures
+and also increases the performance for workload deployments.
+
+### Goals
+
+To make classification more useful by incorporating additional parameters
+to get the matching host. It will help user to specify more rules by
+introducing new parameters.
+
+We will carry out testing of HWCC on Dell hardware.
+
+## Proposal
+
+We are submitting proposal based on the issue raised in HWCC:
+[Issue 53](https://github.com/metal3-io/hardware-classification-controller/issues/53)
+
+* In Disk, supported parameters are only size and count,
+  in addition to these user can identify types of Disk (for eg:
+  HDDs, SSDs, RAID or NVMes). This will help the user for Day-1,
+  Day-2 scenarios.
+* CPU architecture will be useful if the user wants to have 32-bit
+  or 64-bit processor as per his requirements.
+  If user is interested in particular NIC vendor, then NIC vendor
+  can be added as parameter.
+
+* New parameters such as Firmware and SystemVendor will be added
+  advantage to identify matched hosts.
+  System Vendor will be useful to add as a new parameter if the
+  user is looking for particular vendor and their hardware for his
+  application to support or deploy on.
+  Firmware will be useful to add as a new parameter if user is looking
+  for specific feature/s in particular releases or versions.
+
+### Investigation Details
+
+Add support for new parameters:
+
+1. CPU
+
+   `arch` will come under CPU parameter.
+
+   User can provide `arch` type based on requirement.
+
+   Investigation Details: Below is sample introspection data for CPU.
+   `arch` parameter will be added under `cpu` section.
```

Comment:
The arch parameter is already part of the data collected during inspection.

```suggestion
```

Reply:
Done.